__tests__/helper.spec.js

```diff
@@ -1,18 +1,167 @@
+};
+
+describe.only('Helper', () => {
 	describe('hslToRgb should', () => {
 		it('return rgb', () => {
 			// Arrange:
 			const hue = 0.77;
 			const saturation = 0.9;
 			const lightness = 0.8;
 
-			//Act:
+			// Act:
 			const rgb = Helper.hslToRgb(hue, saturation, lightness);
 
-			//Assert:
+			// Assert:
 			expect(rgb).toEqual({R: 215, G: 158, B: 250});
 		});
 	});
+
+	describe('mosaicsFieldObjectBuilder should', () => {
+		beforeEach(() => {
+			stub(MosaicService, 'getMosaics').resolves(Promise.resolve([
+				TestHelper.mockMosaicInfo(mockTestMosaic.idHex, 'TC46AZWUIZYZ2WVGLVEZYNZHSIFAD3AFDPUJMEA', 10, 0)
+			]));
+
+			stub(NamespaceService, 'getMosaicsNames').resolves(Promise.resolve([{
+				names: [new NamespaceName(new NamespaceId(mockTestMosaic.namespaceName), mockTestMosaic.namespaceName)],
+				mosaicId: mockTestMosaic.idHex
+			}]));
+
+			stub(NamespaceService, 'getLinkedMosaicId').resolves(Promise.resolve(new MosaicId(mockTestMosaic.idHex)));
+		});
+
+		afterEach(restore);
+
+		it('returns basic mosaic object', async () => {
+			// Arrange:
+			const mockMosaics = [
+				new Mosaic(new MosaicId(http.networkCurrency.mosaicId), UInt64.fromUint(1)),
+				new Mosaic(new MosaicId(mockTestMosaic.idHex), UInt64.fromUint(4))
+			];
+
+			// Act:
+			const result = await Helper.mosaicsFieldObjectBuilder(mockMosaics);
+
+			// Assert:
+			expect(result).toStrictEqual([
+				{
+					amount: '0.000001',
+					id: new MosaicId(http.networkCurrency.mosaicId).id,
+					mosaicAliasName: http.networkCurrency.namespaceName,
+					mosaicId: http.networkCurrency.mosaicId,
+					rawAmount: UInt64.fromUint(1)
+				},
+				{
+					id: new MosaicId(mockTestMosaic.idHex).id,
+					amount: '4',
+					rawAmount: UInt64.fromUint(4),
+					mosaicId: mockTestMosaic.idHex,
+					mosaicAliasName: [mockTestMosaic.namespaceName]
+				}
+			]);
+		});
+
+		it('returns mosaics object with total amount when mosaic id is same', async () => {
+			// Arrange:
+			const mockMosaics = [
+				new Mosaic(new MosaicId(http.networkCurrency.mosaicId), UInt64.fromUint(1)),
+				new Mosaic(new NamespaceId(http.networkCurrency.namespaceName), UInt64.fromUint(2)),
+				new Mosaic(new MosaicId(mockTestMosaic.idHex), UInt64.fromUint(4)),
+				new Mosaic(new NamespaceId(mockTestMosaic.namespaceName), UInt64.fromUint(2))
+			];
+
+			// Act:
+			const result = await Helper.mosaicsFieldObjectBuilder(mockMosaics);
+
+			// Assert:
+			expect(result).toStrictEqual([
+				{
+					id: new MosaicId(http.networkCurrency.mosaicId).id,
+					amount: '0.000003',
+					mosaicAliasName: http.networkCurrency.namespaceName,
+					mosaicId: http.networkCurrency.mosaicId,
+					rawAmount: UInt64.fromUint(3)
+				},
+				{
+					id: new MosaicId(mockTestMosaic.idHex).id,
+					amount: '6',
+					rawAmount: UInt64.fromUint(6),
+					mosaicId: mockTestMosaic.idHex,
+					mosaicAliasName: [mockTestMosaic.namespaceName]
+				}
+			]);
+		});
+	});
+
+	describe('resolveMosaicId should', () => {
```

Comment:
add test when mosaic being looked up is network currency mosaic

if possible, check if `NamespaceService.getLinkedMosaicId` mocked function is called or not